apps/examples/testcase/ta_tc/st_things/stc/stc_st_things_main.c

```diff
@@ -0,0 +1,227 @@
+extern bool handle_set_request_on_dimming(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_get_request_on_ct(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_ct(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+
+#ifdef CONFIG_RESET_BUTTON
+static bool check_reset_button_pin_number(void)
+{
+	printf("Current pin number : %d\n", CONFIG_RESET_BUTTON_PIN_NUMBER);
+	return CONFIG_RESET_BUTTON_PIN_NUMBER >= 0 ? true : false;
+}
+
+static void gpio_callback_event(void *user_data)
+{
+	printf("gpio_callback_event!!\n");
+	printf("reset :: %d\n", st_things_reset());
+}
+#endif
+
+#ifdef CONFIG_LIGHT_SWITCH_BUTTON
+static void gpio_callback_event_trigger_light_switch(void *user_data)
+{
+	printf("gpio_callback_event to notify switch resource's observers manually!!\n");
+	change_switch_value();
+	printf("Notify to observer :: %d\n", st_things_notify_observers(g_res_switch));
+	set_rep_push_mesg();
+}
+#endif
+
+static void handle_reset_result(bool result)
+{
+	printf("[%s]Reset %s.\n", result ? "succeeded" : "failed", TAG);
+}
+
+static void handle_things_status_change(st_things_status_e things_status)
+{
+	printf("[%s]Things status is changed: %d\n", TAG, things_status);
+}
+
+static bool handle_reset_request(void)
+{
+	printf("[%s]Received a request for RESET.\n", TAG);
+	return true;
+}
+
+static bool handle_ownership_transfer_request(void)
+{
+	printf("[%s]Received a request for Ownership-transfer. \n", TAG);
+	return true;
+}
+
+static bool handle_get_request(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a GET request on %s\n", req_msg->resource_uri);
+
+	if (0 == strncmp(req_msg->resource_uri, g_res_switch, strlen(g_res_switch))) {
+		return handle_get_request_on_switch(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_dimming, strlen(g_res_dimming))) {
+		return handle_get_request_on_dimming(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_temp, strlen(g_res_temp))) {
+		return handle_get_request_on_ct(req_msg, resp_rep);
+	} else {
+		printf("Not supported uri.\n");
+	}
+	return false;
+}
+
+static bool handle_set_request(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+	printf("Received a SET request on %s\n", req_msg->resource_uri);
+
+	if (0 == strncmp(req_msg->resource_uri, g_res_switch, strlen(g_res_switch))) {
+		return handle_set_request_on_switch(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_dimming, strlen(g_res_dimming))) {
+		return handle_set_request_on_dimming(req_msg, resp_rep);
+	} else if (0 == strncmp(req_msg->resource_uri, g_res_temp, strlen(g_res_temp))) {
+		return handle_set_request_on_ct(req_msg, resp_rep);
+	}
+	return false;
+}
+
+int ess_process(void)
+{
+#ifdef CONFIG_RESET_BUTTON
```

Comment:
I guess this function is not required. Please remove all unused functions.

Reply:
Yes removed.